### PR TITLE
Client gobs

### PR DESCRIPTION
**Pull request by jacksonargo**
* Upload server accepts gob encoding (via media-type `application/x.vvgo.pkg.api.uploads.gob`) and gzip encoding
* Client makes requests to server using gob+gzip
* Server does not respond gzip, since nginx does that for us.
* New parts links are added with a revision.
* Client makes async requests to api server

### CODE REVIEW

File: pkg/api/upload.go [hunk 1]
```diff
@@ -110,14 +113,46 @@ func (x UploadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+			logger.WithError(err).Error("gzipReader.Close() failed")
+			badRequest(w, "")
+		}
+
+	default:
+		if _, err := body.ReadFrom(r.Body); err != nil {
+			logger.WithError(err).Error("gzip.Read() failed")
+			badRequest(w, "")
+		}
 	}
```
Comment (0az) [inline pkg/api/upload.go:138]: Is it really necessary to fail on no gzip?

File: pkg/api/upload.go [hunk 2]
```diff
@@ -110,14 +113,46 @@ func (x UploadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
 
 	var documents []Upload
-	if ok := jsonDecode(r.Body, &documents); !ok {
-		badRequest(w, "")
+	switch r.Header.Get("Content-Type") {
+	case "application/octet-stream":
+		if ok := gobDecode(&body, &documents); !ok {
+			badRequest(w, "")
+			return
+		}
```
Comment (0az) [inline pkg/api/upload.go:146]: Maybe we should use a custom MIME type

File: pkg/api/upload.go (hunk 1 above)
Comment (jacksonargo) [reply pkg/api/upload.go:138]: I oh I think I need to just update the log message here. This isn't actually failing if not gzip.

File: pkg/api/upload.go (hunk 2 above)
Comment (jacksonargo) [reply pkg/api/upload.go:146]: Do you have one you recommend?

Comment (0az) [reply pkg/api/upload.go:146]: Maybe `application/x.vvgo.gob-upload` or `application/x.vvgo.upload`.

Ref: https://tools.ietf.org/html/rfc6838#section-3.4

Comment (jacksonargo) [reply pkg/api/upload.go:146]: :ok: Lets use that!

Comment (jacksonargo) [reply pkg/api/upload.go:146]: :thinking: Actually, lets us the golang pkg namespacing for this? Idk... I guess it doesn't really matter too much right now.

**Pull request merged**